customer_satisfaction/README.md

```diff
@@ -0,0 +1,90 @@
+
+As a result of this discussion, we decided to build a pipeline to predict the customer satisfaction score for the next order or purchase. We gave special consideration to ZenML standard pipeline and its steps, MLflow tracking to track our metrics and parameters, and MLflow deployment to deploy our model. 
+
+Our standard training pipeline consists of several steps: 
+
+* ingest_data:- This step will ingest the data from the dataset and create a data frame.
+* clean_data:- This step will clean the data and remove the unwanted columns. 
+* model_train:- This step will train the model and save the model using MLflow auto logging. 
+* evaluation:- This step will evaluate the model and save the metrics using mlfow autlogging into the artifact store.  
+
+### Depoloyment & Inference Pipeline 
+
+We have another pipeline, the deployment_pipeline.py, that implements a continuous deployment workflow. It ingests and processes input data, trains a model and then (re)deploys the prediction server that serves the model if it meets some evaluation criteria.
+
+In the deployment pipeline, ZenML's MLflow tracking integration is used for logging the hyperparameter values and the trained model itself and the model evaluation metrics -- as MLflow experiment tracking artifacts into the local MLflow backend. This pipeline also launches a local MLflow deployment server to serve the latest MLflow model if its accuracy is above a configured threshold.
+
+The MLflow deployment server runs locally as a daemon process that will continue to run in the background after the example execution is complete.
+
+The deployment pipeline has caching enabled to avoid re-training the model if the training data and hyperparameter values don't change. When a new model is trained and passes the accuracy threshold validation, the pipeline automatically updates the currently running MLflow deployment server to serve the new model instead of the old one.
+
+We also have an inference pipeline that interacts with the continuous prediction server deployed to get online predictions based on live data. The inference pipeline simulates loading data from a dynamic external source and then uses that data to perform online predictions using the running MLflow prediction server.
+
+![training_and_deployment_pipeline](_assets/training_and_deployment_pipeline.png)
+
+## :notebook: Diving into the code
+
+You have two pipelines to run, `run_pipeline.py`, a traditional ZenML Pipeline, and `run_deployment.py`, a continuous deployment pipeline. So, if you want to run `run_pipeline.py`, you can run the following command: 
+
+```bash
+python run_pipeline.py
+```
+
+And this will run the standard ZenML pipeline. If you want to run `run_deployment.py`, you can run the following command:
+
+```bash
+python run_deployment.py
+```
+
+## :question: FAQ
+
+1. When running the continuous deployment pipeline, I get an error about No Step found for the name `mlflow_deployer`.
```

Comment:
```suggestion
1. When running the continuous deployment pipeline, I get an error about No Step found for the name `mlflow_deployer`.
```

Maybe instead of "I get an error about No Step found"... you could specify what error gets raised?

Reply:
Sure